### PR TITLE
Got annotate_from_gff working from gff3 files

### PR DESCRIPTION
**Pull request by ChristopherBradley**
New pull request for the develop branch

### CODE REVIEW

File: src/cogent3/parse/gff.py [hunk 1]
```diff
@@ -14,27 +14,43 @@
 __email__ = "pm67nz@gmail.com"
 __status__ = "Production"
 
-from io import StringIO
 from pathlib import Path
 
 from cogent3.util.misc import open_
+from cogent3.util.union_dict import UnionDict
```
Comment (ChristopherBradley) [inline src/cogent3/parse/gff.py]: unused import

File: src/cogent3/parse/gff.py [hunk 2]
```diff
@@ -45,46 +61,70 @@ def gff2_parser(f):
...
+        }
+        yield rtn
+
+
+def parse_attributes_gff2(attributes, span):
+    """Returns a dict with name and info keys"""
+    name = attributes[attributes.find('"') + 1 :]
+    if '"' in name:
+        name = name[: name.find('"')]
+    id_ = name if name else str(span)
```
Comment (GavinHuttley) [inline src/cogent3/parse/gff.py]: span not safe as multiple distinct features can map to same coords. We need to construct a fake identifier with a count.

```python
class gff_attr_parser:
      def __init__(self, gff3):
        # define attributes
      self_num_fake = 0
       self._gff3 = gff3
    def __call__(self, attrs):
       if self._gff3:
        pass
```

File: src/cogent3/parse/gff.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
     "Christopher Bradley",
 ]
 __license__ = "BSD-3"
-__version__ = "2019.11.15.a"
+__version__ = "2019.10.24a"
```
Comment (GavinHuttley) [inline src/cogent3/parse/gff.py]: global search for this string `"2019.10.24a"` and replace with `"2019.11.15.a"`

File: tests/test_core/test_alignment.py [hunk 4]
```diff
@@ -899,6 +900,42 @@ def test_annotate_from_gff(self):
...
+        self.assertEqual(len(matches), 14)
+        matches = [m for m in aln_seq.get_annotations_matching("gene")]
+        self.assertEqual(len(matches), 1)
+        matches = matches[0].get_annotations_matching("mRNA")
+        self.assertEqual(len(matches), 1)
+        matches = matches[0].get_annotations_matching("exon")
+        self.assertEqual(len(matches), 3)
+
+        # Class doesn't matter, but only need to run this once
+        if self.Class == Alignment:
```
Comment (GavinHuttley) [inline tests/test_core/test_alignment.py]: move this test to `test_sequence`

**Pull request merged**